### PR TITLE
Updated pymc.simulator docstring (typos, defaults, type description)

### PR DESCRIPTION
**Pull request by daniel-saunders-phil**
**What is this PR about?**

fixing a number of issues with the pymc.simulator docstring.

...

**Checklist**
+ [ ] Explain important implementation details 👆
+ [ ] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [ ] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [ ] Are the changes covered by tests and docstrings?
+ [ ] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- ...  added defaults to each parameter, changed instances of Aesara Op to correct format Aesara_Op, fixed a typo

### CODE REVIEW

File: pymc/distributions/simulator.py [hunk 1]
```diff
@@ -98,20 +98,18 @@ class Simulator(NoDistribution):
...
         ``distance="laplace"`` + ``sum_stat="sort"`` is equivalent to the the 1D 1-wasserstein distance
-    sum_stat : Aesara Op, callable or str
-        Summary statistic function. Available options are ``"indentity"`` (default),
-        ``"sort"``, ``"mean"``, ``"median"``. If a callable (or Aesara Op) is defined,
+    sum_stat : Aesara_Op, callable or str, default "identity"
+        Summary statistic function. Available options are ``"identity"``,
+        ``"sort"``, ``"mean"``, ``"median"``. If a callable (or Aesara_Op) is defined,
         it should return a 1d numpy array (or Aesara vector).
-    epsilon : float or array
+    epsilon : float or array, default 1.0
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]: I think this should be `tensor_like of float`

File: pymc/distributions/simulator.py [hunk 2]
```diff
@@ -98,20 +98,18 @@ class Simulator(NoDistribution):
...
         Scaling parameter for the distance functions. It should be a float or
-        an array of the same size of the output of ``sum_stat``. Defaults to ``1.0``
-    ndim_supp : int
+        an array of the same size of the output of ``sum_stat``.
+    ndim_supp : int, default 0
         Number of dimensions of the SimulatorRV (0 for scalar, 1 for vector, etc.)
-        Defaults to ``0``.
-    ndims_params : list[int]
+    ndims_params : list[int], default 0 for each parameter.
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
    ndims_params : list of int, optional
```

As the default is a bit complex here, it is not a value we can add here but something that depends on other parameters, I would use optional and keep the explanation in the description

File: pymc/distributions/simulator.py [hunk 3]
```diff
@@ -83,9 +83,9 @@ class Simulator(NoDistribution):
         be passed by order, in which case the keyword argument ``params`` is
         ignored. Alternatively, each parameter can be passed by order after fn,
         ``param1, param2, ..., paramN``
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]: Can't comment further up. It looks like `unnamed_params` is missing. I think it should be something like

```
*unnamed_params : list of TensorVariable
    description starts here
params : list of TensorVariable
    description extended here, plus mention that if both are present an error is thrown.
```

someone else will need to take over on the actual description though as I don't really know what to write. Is one preferred over the other, are there any potential behaviour differences? 

If they are both exactly the same, then the we can follow for example xarray's pattern where this situation is very common with kwargs, see for example: https://docs.xarray.dev/en/stable/generated/xarray.Dataset.rename.html

**Review by OriolAbril — COMMENTED** [review]
Thanks!

It already looks great, and the docs look better and are more clear already. I left some more comments, plus this one here because it is on a line I wasn't able to comment on: the short description also has

```
``pm.sample_smc``
# should instead be
:func:`~pymc.sample_smc`
```

**Comment by daniel-saunders-phil** [comment]
Hi Oriol, thanks for the feedback - these suggestions are really helpful. I'm a bit unsure about the procedure. To revise a pull request, do I make the changes on my branch and then resubmit the whole PR through the git command line instructions or is there a different procedure?

**Comment by OriolAbril** [comment]
Make the changes locally and commit them to the branch you opened your PR from, `docstring_update` in this case. After that you only need to push the changes and this same PR will be updated automatically with the new commits.

This is the last section in https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/pr_tutorial.html#get-ready-for-reviews which has the specific commands and extends the overview I gave in the above paragraph.

File: pymc/distributions/simulator.py (hunk 3 above)
Comment (daniel-saunders-phil) [reply pymc/distributions/simulator.py]: It looks like they should behave identically. If the keyword argument isn't passed, then params just gets defined as the unnamed_params and then params is used throughout the rest of the file. I went ahead and wrote a description similar to xarray.

**Comment by daniel-saunders-phil** [comment]
I implemented the suggested changes and then pushed them. 

Also, one thing came up that wasn't strictly a docstring fix. I noticed the file uses inconsistent spellings of Kullback-Leiber and the class is incorrectly called KullbackL**ie**bler. I thought we might want to change all of them to the correct spelling. I searched through the rest of the pymc package and couldn't find any references to the incorrectly spelled class, so I don't think this should cause trouble for other functions. But I'm also happy to ignore this and just stick to docstrings for now.

**Comment by reshamas** [comment]
@daniel-saunders-phil 
I think it's ok to fix the spelling of Kullback-Leiber.
I've been fixing spelling in the notebooks I have been updating, and it's been ok.

**Comment by daniel-saunders-phil** [comment]
I’m not sure how to handle this pre-commit problem. I’m gonna try to figure it out for a little bit but I might have some questions in a couple of days.

**Comment by reshamas** [comment]
@daniel-saunders-phil I am working on my 3rd notebook and, once again(!), stuck at the pre-commit stage. (+ possible git issue)

@OriolAbril 
Perhaps we should have a special office hours session on how to get through pre-commit.  ;)

cc:  @symeneses

**Comment by OriolAbril** [comment]
@daniel-saunders-phil if you click on the "Details" button of the checks seciton you'll be able to see the checks logs. In this PR, the issue is with the "trim trailing whitespace": https://github.com/pymc-devs/pymc/runs/7715083643?check_suite_focus=true#step:4:64.

That means that there are lines that end up with a white character instead of finishing with the last letter/punctiation sign of the line. If you have followed the steps from https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/environment_setup.html to set up your environment, you can run `pre-commit run --all` on your command line (while inside the pymc-dev environment). pre-commit will fail, but it will also modify the files and remove those trailing whitespaces. You can then git add and git commit those changes and push, which should fix the CI here.

**Comment by daniel-saunders-phil** [comment]
It looks like all tests are passing this time! after a little bit of tinkering, I managed to get pre-commit working. Thanks a lot, @OriolAbril and @reshamas !

File: pymc/distributions/simulator.py [hunk 4]
```diff
@@ -66,7 +66,7 @@ def sum_stat(cls, *args, **kwargs):
 class Simulator(NoDistribution):
     r"""
     Simulator distribution, used for Approximate Bayesian Inference (ABC)
-    with Sequential Monte Carlo (SMC) sampling via ``pm.sample_smc``.
+    with Sequential Monte Carlo (SMC) sampling via ``~pymc.sample_smc``.
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
    with Sequential Monte Carlo (SMC) sampling via :func:`~pymc.sample_smc`.
```

it needs to use this syntax in order to be rendered as a link, not only use the correct import path of the function. Ref: https://pymc-data-umbrella.xyz/en/latest/about/contributing_to_documentation/docs_presentation.html#cross-referencing-roles

File: pymc/distributions/simulator.py [hunk 5]
```diff
@@ -78,14 +78,20 @@ class Simulator(NoDistribution):
...
-    distance : Aesara Op, callable or str
-        Distance function. Available options are ``"gaussian"`` (default), ``"laplace"``,
-        ``"kullback_leibler"`` or a user defined function (or Aesara Op) that takes
+
+    *unnamed_params : list of TensorVariable
+        Parameters used by the Simulator random function. Each parameter can be passed
+        by order after fn, for example ``param1, param2, ..., paramN``. params can also
+        be passed with keyword argument "params".
+
+    params : list of TenssorVariable
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
    params : list of TensorVariable
```

File: pymc/distributions/simulator.py [hunk 6]
```diff
@@ -78,14 +78,20 @@ class Simulator(NoDistribution):
...
         and ``size`` defines the size of the desired sample.
-    params : list
-        Parameters used by the Simulator random function. Parameters can also
-        be passed by order, in which case the keyword argument ``params`` is
-        ignored. Alternatively, each parameter can be passed by order after fn,
-        ``param1, param2, ..., paramN``
-    distance : Aesara Op, callable or str
-        Distance function. Available options are ``"gaussian"`` (default), ``"laplace"``,
-        ``"kullback_leibler"`` or a user defined function (or Aesara Op) that takes
+
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
```

File: pymc/distributions/simulator.py [hunk 7]
```diff
@@ -78,14 +78,20 @@ class Simulator(NoDistribution):
...
+    *unnamed_params : list of TensorVariable
+        Parameters used by the Simulator random function. Each parameter can be passed
+        by order after fn, for example ``param1, param2, ..., paramN``. params can also
+        be passed with keyword argument "params".
+
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
```

File: pymc/distributions/simulator.py [hunk 8]
```diff
@@ -78,14 +78,20 @@ class Simulator(NoDistribution):
...
+    params : list of TenssorVariable
+        Keyword form of ''unnamed_params''.
+        One of unnamed_params or params must be provided.
+        If passed both unnamed_params and params, an error is raised.
+
```
Comment (OriolAbril) [inline pymc/distributions/simulator.py]:
```suggestion
```

**Review by OriolAbril — APPROVED** [review]

**Review by michaelosthege — APPROVED** [review]

**Review by michaelosthege — CHANGES_REQUESTED** [review]
Sorry for requesting changes this late, but I think we should keep an alias of the `KullbackLeibler` class:

```python
class KullbackLiebler(KullbackLeibler):
    def __init__(self,  *args, **kwargs):
        """Alias for KullbackLeibler."""
        warnings.warn("You are using `KullbackLiebler` which is a typo. Use `KullbackLeibler` instead.", DeprecationWarning)
        super(). __init__(*args, **kwargs)

**Comment by OriolAbril** [comment]
I considered it too but but eventually decided on not doing it because 1) it is not included in our docs and therefore not part of the public api and 2) no tests broke because it has not been refactored to v4,  so it would be a breaking change on the private api with a major change version, from 3.11 to 4.x (whatever the version it gets updated).

We can still add the alias to be nice,  but then it should have a different type of warning. This is the description of deprecationwarning in https://docs.python.org/3/library/exceptions.html#DeprecationWarning :

> Base class for warnings about deprecated features when those warnings are intended for other Python developers.
>
> Ignored by the default warning filters, except in the __main__ module (PEP 565). Enabling the Python Development Mode shows this warning.

**Review by michaelosthege — APPROVED** [review]
ok, thanks Oriol! I didn't realize this wasn't public API.

**Comment by reshamas** [comment]
Congrats @daniel-saunders-phil !

#DataUmbrellaPyMCSprint

**Pull request merged**